app/code/core/Mage/ConfigurableSwatches/Helper/Mediafallback.php

```diff
@@ -194,36 +194,38 @@ public function getConfigurableImagesFallbackArray(
             }, $mediaGallery['images']);
 
             // load images from the configurable product for swapping
-            foreach ($mapping as $map) {
-                $imagePath = null;
-
-                //search by store-specific label and then default label if nothing is found
-                $imageKey = array_search($map['label'], $imageHaystack);
-                if ($imageKey === false) {
-                    $imageKey = array_search($map['default_label'], $imageHaystack);
-                }
-
-                //assign proper image file if found
-                if ($imageKey !== false) {
-                    $imagePath = $mediaGallery['images'][$imageKey]['file'];
-                }
-
-                $imagesByLabel[$map['label']] = [
-                    'configurable_product' => [
-                        Mage_ConfigurableSwatches_Helper_Productimg::MEDIA_IMAGE_TYPE_SMALL => null,
-                        Mage_ConfigurableSwatches_Helper_Productimg::MEDIA_IMAGE_TYPE_BASE => null,
-                    ],
-                    'products' => $map['product_ids'],
-                ];
+            if (is_iterable($mapping)) {
```

Comment:
We expect `$mapping` to be an array:
```suggestion
            if (is_array($mapping)) {
```

Reply:
genuine question, if we expect it to be an array, do we have to check if it's an array?

Reply:
The code that sets the mapping on the product is skipped based on this condition, so it can end up not getting set:
https://github.com/OpenMage/magento-lts/blob/b67a8e54816d529b620f76b48c66d0e9e0447159/app/code/core/Mage/ConfigurableSwatches/Helper/Mediafallback.php#L69-L71

Another solution would be to use null coalescing operator to avoid the `is_iterable` or `is_array` check:
```patch
- $mapping = $product->getChildAttributeLabelMapping();
+ $mapping = $product->getChildAttributeLabelMapping() ?? [];
```

Reply:
I think both of the things should be implemented :-)